File: wlr/SPX.Text/DefaultHudText.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using SPX.Core;
using System;

namespace SPX.Text
{
    [Serializable]
    class DefaultHudText:Text
    {
        private Color _color;
        private Vector2 _origin;

        public DefaultHudText(float alpha=1f)
        {
            _color = new Color(255f, 255f, 255f, alpha);
        }

        public DefaultHudText(string contents, int x, int y,Vector2 origin,float alpha=1f)
            : base(contents, x, y,TextType.Inventory)
        {
            _color = new Color(255f,255f,255f,alpha);
            _origin = origin;
        }

        public void Reset(string contents, int x, int y,Vector2 origin)
        {
            _origin = origin;
            Reset(contents, x, y);
        }

        public override int Update()
        {
           return 1;
        }
        public override void Draw()
        {
            XnaManager.Renderer.DrawString(TextManager.GetFont(), _contents, _position+_origin, _color, 0, Vector2.Zero,.75f, SpriteEffects.None, Depth.DefaultHudText);
        }
    }
}

File: code/spx-ported/src/com/spx/text/DefaultHudText.java
package com.spx.text;import java.util.*;import com.spx.core.*;import com.xna.wrapper.*;
    class DefaultHudText extends Text
    {
        private Color _color;
        private Vector2 _origin;

        public DefaultHudText(float alpha)
        {
            _color = new Color(255f, 255f, 255f, alpha);
        }

        public DefaultHudText(String contents, int x, int y,Vector2 origin,float alpha)
            {        	super(contents, x, y,TextType.Inventory);            _color = new Color(255f,255f,255f,alpha);
            _origin = origin;
        }

        public void Reset(String contents, int x, int y,Vector2 origin)
        {
            _origin = origin;
            Reset(contents, x, y);
        }
        @Override
        public int Update()
        {
           return 1;
        }                @Override
        public void Draw()
        {
            XnaManager.Renderer.DrawString(TextManager.GetFont(), _contents, _position.add(_origin), _color, 0, Vector2.Zero,.75f, SpriteEffects.None, Depth.DefaultHudText);
        }
    }
